src/Microsoft.AspNet.Routing/RouteBuilder.cs

```diff
@@ -21,6 +23,14 @@ public RouteBuilder(IApplicationBuilder applicationBuilder, IRouter defaultHandl
                 throw new ArgumentNullException(nameof(applicationBuilder));
             }
 
+            if (applicationBuilder.ApplicationServices.GetService(typeof(RoutingMarkerService)) == null)
+            {
+                throw new InvalidOperationException(Resources.FormatUnableToFindServices(
```

Comment:
Hmm - what's the reason for this?  Like, what's the callstack that needs to throw that isn't throwing?